tools/accuracy_checker/accuracy_checker/representation/base_representation.py

```diff
@@ -46,3 +46,6 @@ def set_segmentation_mask_source(self, mask_source):
 
     def set_additional_data_source(self, source):
         self.metadata['additional_data_source'] = source
+
+    def set_dataset_metadata(self, source):
+        self.metadata.update(source)
```

Comment:
on my view better to add key `dataset_meta` and put in this field, in your solution, possible intersection and overriding original meta keys on datasets meta values